cmd/kubectl-gadget/sigsnoop.go

```diff
@@ -60,7 +60,12 @@ var sigsnoopCmd = &cobra.Command{
 			},
 		}
 
-		return utils.RunTraceAndPrintStream(config, sigsnoopTransformLine)
+		err := utils.RunTraceAndPrintStream(config, sigsnoopTransformLine)
+		if err != nil {
+			return utils.WrapInErrRunGadget(err)
+		}
+
+		return nil
```

Comment:
Absolutely. There is no advantage of wrapping the error on the caller but doing it at every return point of the functions themselves does not seem to be a clean solution either. In particular because, as you said, most of those functions have multiple return points.

With @mauriciovasquezbernal, we were discussing that we need to think about a refactorization on the gadgets because there is a lot of repeated code and this is another example.

If it is ok for you, I will keep this PR as it is but create an issue to improve it.

Reply:
Sure, yes, let's do it in a separate PR.

Reply:
I created issue https://github.com/kinvolk/inspektor-gadget/issues/545 to track this task.